hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/RecoveredReplicationSourceShipper.java

```diff
@@ -92,10 +86,7 @@ private void terminate(String reason, Exception cause) {
         "Closing worker for wal group " + this.walGroupId + " because an error occurred: " + reason,
         cause);
     }
-    entryReader.interrupt();
-    Threads.shutdown(entryReader, sleepForRetries);
-    this.interrupt();
-    Threads.shutdown(this, sleepForRetries);
-    LOG.info("ReplicationSourceWorker {} terminated", this.getName());
+    LOG.info("ReplicationSourceWorker stopping");
```

Comment:
nit: move into stopWorker()?

Reply:
Nah.  Better where it is so that, through logging, it is known what other class is stopping the thread.  Also, it could be very verbose if somewhere in the code it is repeatedly calling the method.